src/main/java/org/broadinstitute/hellbender/tools/exome/IntervalBackedExonDB.java

```diff
@@ -0,0 +1,361 @@
+        if (intervals.isEmpty())
+            return;
+        int nextIndex = 0;
+        GenomeLoc lastLocation = null;
+        for (final GenomeLoc location : intervals) {
+            checkInputIntervalCoherence(lastLocation, location);
+            final String name = intervalName(nextIndex,location);
+            if (name != null) {
+                final GenomeLoc previous = intervalsByName.put(name,location);
+                if (previous != null)
+                    throw new IllegalStateException("more than one interval results in the same name (" +
+                            name + "): " + previous + " and "  + location);
+            }
+            sortedIntervals.add(location);
+            nextIndex++;
+            lastLocation = location;
+        }
+    }
+
+    /**
+     * Throws a exception if there some interval location incoherence.
+     *
+     * @param lastLocation previous location.
+     * @param nextLocation next location to add.
+     *
+     * @throws IllegalArgumentException if there is some problem with a location.
+     */
+    private void checkInputIntervalCoherence(final GenomeLoc lastLocation, final GenomeLoc nextLocation) {
+        if (nextLocation == null)
+            throw new IllegalArgumentException("the input intervals contain a null location");
+        if (nextLocation.isUnmapped())
+            throw new IllegalArgumentException("the input intervals contain unmapped locations");
+        if (nextLocation.equals(GenomeLoc.WHOLE_GENOME))
+            throw new IllegalArgumentException("the input intervals contain a whole-genome location");
+        if (lastLocation != null && !lastLocation.isBefore(nextLocation))
+            throw new IllegalArgumentException(
+                    "the input intervals list is not in order or " +
+                            "contains overlapped intervals: "
+                            + lastLocation + " should be before " + nextLocation);
+    }
+
+    /**
+     * Produces the unique name for a genomic interval.
+     *
+     * <p>This name can depend in either the index of the interval, its location or both</p>
+     *
+     * <p>The implementation can assume that input interval is a proper one with start and stop positions and a contig,
+     * thus the result is unpredictable with special intervals values such as {@link GenomeLoc#UNMAPPED}
+     *   or {@link GenomeLoc#WHOLE_GENOME}.
+     *
+     * <p>
+     *     The implementation of this method is such that the name will be unique, i.e. that two different locations
+     *     won't ever result in the same name.
+     * </p>
+     *
+     * @param index the final index of the interval in the exome.
+     *
+     * @param location the genome location of the interval.
+     *
+     * @return can return {@code null} indicating that this interval should remain anonymous.
+     */
+    protected String intervalName(@SuppressWarnings("unused") final int index, final GenomeLoc location) {
```

Comment:
if you don't use the variable yourself, don't add it. It's easy to add later. Let's not write code that is immediately unused.